cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,204 @@
+
+    # the install root is created
+    assert context.install_root.is_dir(), f"Invalid install root {context.install_root}"
+
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer_fixture, context, filename):
+    """ Installer key binaries exist. """
+    assert (context.engine_bin_path / filename).is_file(), f"{filename} not found in {context.engine_bin_path}"
+
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(test_installer_fixture, context):
+    """ Engine is registered when o3de.exe is run. """
+    try:
+        result = context.run([str(context.engine_bin_path / 'o3de.exe')], timeout=15)
+
+        # o3de.exe should not close with an error code 
+        assert result.returncode == 0, f"o3de.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        # we expect to close the app ourselves
+        pass
+
+    # a valid engine.json exists
+    engine_json_data = manifest.get_engine_json_data(engine_name=None, engine_path=context.install_root)
+    assert engine_json_data, f"Failed to get engine.json data for engine in {context.install_root}"
+    assert 'engine_name' in engine_json_data, "Engine.json does not contain engine_name key"
+
+    # the engine is registered
+    engine_name = engine_json_data['engine_name']
+    engine_registered_path = manifest.get_registered(engine_name = engine_name)
+
+    assert engine_registered_path, f"Failed to get registered engine path for {engine_name}"
+    assert engine_registered_path.resolve() == context.install_root, f"{engine_name} is registered to {engine_registered_path} instead of {context.install_root}"
+
+@pytest.fixture(scope="session")
+def test_create_project_fixture(test_o3de_registers_engine_fixture, context):
+    """ o3de.bat CLI creates a project. """
+    o3de_path = context.install_root / 'scripts/o3de.bat'
+    result = context.run([str(o3de_path),'create-project','--project-path', str(context.project_path)])
+    assert result.returncode == 0, f"o3de.bat failed to create a project with exit code {result.returncode}"
+
+    project_json_path = Path(context.project_path) / 'project.json'
+    assert project_json_path.is_file(), f"No project.json found at {project_json_path}"
+
+
+@pytest.fixture(scope="session")
+def test_compile_project_fixture(test_create_project_fixture, context):
+    """ Project can be configured and compiled. """
+    project_name = Path(context.project_path).name
+    cmake_path = next(context.cmake_runtime_path.glob('**/cmake.exe'))
+    launcher_target = f"{project_name}.GameLauncher"
+
+    # configure
+    result = context.run([str(cmake_path),'-B', str(context.project_build_path), '-S', '.', '-G', 'Visual Studio 16 2019'], cwd=context.project_path)
```

Comment:
You may want to store of Generator in a constant at the top of this file, for when we inevitably want to test to VS2022

Reply:
I think I might tackle that in a separate PR that may also tackle PALification.